### ISSUE TITLE
Inconsistencies in AnionCorrection

### ISSUE THREAD

**Issue by wmdrichards** [materialsproject/pymatgen#710]
The application of the anion energy correction is extremely inconsistent when correct_peroxide=True, making it very difficult to tell what is going on.

The correction applied by AnionCorrection depends on whether or not a structure is present in the entry, and whether the oxide_type parameter is present. If neither of these are included in the ComputedEntry, no peroxide/superoxide/ozonide correction is applied unless the composition happens to fall into one of the 'common' peroxide/superoxide/ozonide formulas.

As a result of this confusion, the corrections are often not correctly applied. For example, https://materialsproject.org/materials/mp-2310/ does not have the peroxide correction on the website energy, though if you pass an entry including a structure it correctly recognizes it as a peroxide. It appears as though the only entries on MP that have the peroxide corrections are those in the common formulas list.

## Summary

* AnionCorrection application is inconsistent

## Example code

```python
from pymatgen.entries.compatibility import MaterialsProjectCompatibility
from pymatgen.matproj.rest import MPRester

for e in MPRester('XXXX').get_entries('CdO2', inc_structure=True):
    print MaterialsProjectCompatibility().process_entry(e).correction

for e in MPRester('XXXX').get_entries('CdO2', inc_structure=False):
    print MaterialsProjectCompatibility().process_entry(e).correction
```

## Suggested solution (if any)

* Get rid of the fallback to using common formulas to identify peroxides
* Raise an error when trying to apply a peroxide/superoxide/ozonide correction when no structure or oxide_type attribute is present
* Rebuild the database (since corrected energies will change)

**Comment by shyuep** [comment]
I fixed it by adding oxide_type by default to the entry.data queried from MP. I retained the fallback to special formulas, but added a warning. 

As for DB rebuild, this has to be done by @tschaume and @kristinpersson, and I strongly recommend an additional check in the build validators (@computron and Co.) as well as changing the REST interface by default (@dwinston) to add oxide_type to entry queries.
